File: BringMeHome/BringMeHome.Services/Database/ApplicationStatus.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Services.Database
{
    public class ApplicationStatus
    {
        [Key]
        public int StatusID { get; set; }

        [Required, StringLength(50)]
        public string StatusName { get; set; }

        [StringLength(200)]
        public string Description { get; set; }

        // Navigation properties
        public virtual ICollection<AdoptionApplication> Applications { get; set; }
    }
}

File: bringmehomeUI/bringmehome_android/lib/models/application_status.dart
class ApplicationStatus {
  final int statusID;
  final String statusName;

  ApplicationStatus({required this.statusID, required this.statusName});

  factory ApplicationStatus.fromJson(Map<String, dynamic> json) {
    return ApplicationStatus(
      statusID: json['statusID'] as int,
      statusName: json['statusName'] as String? ?? 'Unknown Status',
    );
  }
}